### PR TITLE
docs: ADR for the purpose of this repo

### PR DESCRIPTION
**Pull request by ormsbee**
First of a few ADRs in this batch.

### CODE REVIEW

File: docs/decisions/0001-purpose-of-this-repo.rst [hunk 1]
```diff
@@ -4,21 +4,33 @@
 Context
 -------
 
-TODO: Give context on what lead to the creation of this repo.
-      What project is the repo related to?
+The code that operates on learning content in Open edX primarily resides in edx-platform, and suffers from a few high-level issues:
+
+* Plugins that want to interact with content effectively edx-platform (a large repo with hundreds of dependencies) as a direct dependency, making development and testing more difficult.
```
Comment (kdmccormick) [inline docs/decisions/0001-purpose-of-this-repo.rst:9]: > ... with content effectively edx-platform (...

I think you accidentally a word here.

**Review by kdmccormick — APPROVED** [review]
Nice 👍🏻

File: docs/decisions/0001-purpose-of-this-repo.rst (hunk 1 above)
Comment (feanil) [inline docs/decisions/0001-purpose-of-this-repo.rst:9]:
```suggestion
* Plugins that want to interact with content effectively import  all of edx-platform (a large repo with hundreds of dependencies) as a direct dependency, making development and testing more difficult.
```

File: docs/decisions/0001-purpose-of-this-repo.rst [hunk 2]
```diff
@@ -4,21 +4,33 @@
...
 Decision
 --------
 
-TODO: Clearly state how the context above led you to creating this repo.
+The openedx-learning repository was created to provide a new place for certain core learning concepts, data models, and APIs to be implemented. These concepts will be more granular and composable than the courses we have today.
+
+This would have two long term goals:
+
+#. Enable learning-related plugins to be built in a simple, maintainable, robust way.
+#. Allow Open edX to support for more flexible and extensible learning experiences at scale.
```
Comment (feanil) [inline docs/decisions/0001-purpose-of-this-repo.rst:20]:
```suggestion
#. Allow Open edX to support more flexible and extensible learning experiences at scale.
```

**Review by feanil — APPROVED** [review]
Couple of grammar suggestions but otherwise looks good.

**Pull request merged**